src/iexamine.cpp

```diff
@@ -1608,7 +1608,7 @@ std::list<item> iexamine::get_harvest_items( const itype &type, const int plant_
     const auto add = [&]( const itype_id &id, const int count ) {
         item new_item( id, calendar::turn );
         if( new_item.count_by_charges() && count > 0 ) {
-            new_item.charges = count;
+            new_item.charges *= count;
```

Comment:
Not in a simple way. Not all `itype` subclasses have the `def_charges` field. The `Item` constructor does a bunch of checks, and sets charges to the default if it's one of the item types with a subclass that has `def_charges`, with dynamic casts. 

Notably, harvest products do *not* have to be comestibles, and non-comestibles can have seeds. As such, we cannot assume the harvest items will be comestibles, and would probably have to do some of the same sorts of checks that he item constructor does, possibly in a separate utility function. So, doable, but more complex.